test/extended/router/gatewayapicontroller.go

```diff
@@ -195,26 +200,7 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 	})
 
 	g.It("Ensure LB, service, and dnsRecord are created for a Gateway object", func() {
-		g.By("Ensure default GatewayClass is accepted")
-		errCheck := checkGatewayClass(oc, gatewayClassName)
-		o.Expect(errCheck).NotTo(o.HaveOccurred(), "GatewayClass %q was not installed and accepted", gatewayClassName)
-
-		g.By("Getting the default domain")
-		defaultIngressDomain, err := getDefaultIngressClusterDomainName(oc, time.Minute)
-		o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
-		defaultDomain := strings.Replace(defaultIngressDomain, "apps.", "gw-default.", 1)
-
-		g.By("Create the default Gateway")
-		gw := names.SimpleNameGenerator.GenerateName("gateway-")
-		gateways = append(gateways, gw)
-		_, gwerr := createAndCheckGateway(oc, gw, gatewayClassName, defaultDomain)
-		o.Expect(gwerr).NotTo(o.HaveOccurred(), "failed to create Gateway")
-
-		g.By("Verify the gateway's LoadBalancer service and DNSRecords")
-		assertGatewayLoadbalancerReady(oc, gw, gw+"-openshift-default")
-
-		// check the dns record is created and status of the published dnsrecord of all zones are True
-		assertDNSRecordStatus(oc, gw)
+		assertGWLBServiceAndDnsrecordProvisioned(oc, gatewayClassName, gateways)
```

Comment:
I'm not sure if this is better that adding everything into one func, if possible you could create another gateway other than the "default" one in the negative test. Please refer to the front part of "Ensure HTTPRoute object is created" test.

Reply:
Thanks for the review. restored and removed the function. Also created a custom gateway for the testing. The example helped too much...